web/profiles/custom/os2loop/modules/os2loop_oembed/src/TwigExtension/TwigExtension.php

```diff
@@ -0,0 +1,254 @@
+    if (!empty($this->findIframe($string))) {
+      $string = $this->handleIframe($string);
+    }
+    $videoArray = $this->createVideoFromUrl($string);
+    $videoArray = $this->applyCookieConsent($videoArray);
+
+    $render = [
+      '#theme' => 'os2loop_video_iframe',
+      '#video' => $videoArray,
+    ];
+
+    return $this->renderer->render($render);
+  }
+
+  /**
+   * List of allowed providers.
+   *
+   * @var array
+   *   The allowed providers.
+   */
+  private const ALLOWED_PROVIDERS = [
+    'vimeo.com' => [
+      'type' => 'Oembed',
+      'provider' => 'Vimeo',
+      'requiredCookies' => 'cookie_cat_statistic',
+    ],
+    'player.skyfish.com' => [
+      'type' => 'custom',
+      'requiredCookies' => NULL,
+    ],
+    'media.videotool.dk' => [
+      'type' => 'custom',
+      'requiredCookies' => NULL,
+    ],
+    'web.microsoftstream.com' => [
+      'type' => 'Oembed',
+      'provider' => 'Microsoft Stream',
+      'requiredCookies' => NULL,
+    ],
+
+  ];
+
+  /**
+   * Search a text for an iframe.
+   *
+   * @param string $text
+   *   The text to search for.
+   *
+   * @return array
+   *   An array representing a match.
+   */
+  private function findIframe($text): array {
+    preg_match('/(?:<iframe[^>]*)(?:(?:\/>)|(?:>.*?<\/iframe>))/', $text, $matches);
+    return $matches;
+  }
+
+  /**
+   * Create an array containing video information.
+   *
+   * @param string $text
+   *   The text input to create video from.
+   *
+   * @return array
+   *   The resulting video array.
+   *
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   *   Exception if oembed fails.
+   */
+  private function createVideoFromUrl($text): array {
+    $video = [];
+    if (filter_var($text, FILTER_VALIDATE_URL)) {
+      $url = parse_url($text);
+      if (array_key_exists($url['host'], static::ALLOWED_PROVIDERS)) {
+        $video['host'] = $url['host'];
+        // Use oembed to create iframe if possible.
+        if ('Oembed' === static::ALLOWED_PROVIDERS[$url['host']]['type']) {
+          try {
+            $url = $this->urlResolver->getResourceUrl($text);
+            $request = $this->httpClient->request('GET', $url);
+            $status = $request->getStatusCode();
+            if (200 == $status) {
+              $video['oembed'] = Json::decode($request->getBody()->getContents());
+              $video['iframe'] = $video['oembed']['html'];
+            }
+          }
+          catch (\Exception $e) {
+            if ('No matching provider found.' === $e->getMessage()) {
+              $this->messenger->addWarning($this->t('Could not build video.'));
+              $video = [];
+            }
+          }
+        }
+        // If oembed is not an option create iframe from a url.
+        elseif ('custom' === static::ALLOWED_PROVIDERS[$url['host']]['type']) {
+          $video['custom']['src'] = $text;
+          $video['iframe'] = '<iframe src="' . $text . '"></iframe>';
```

Comment:
This is a very simple `iframe` template and we may need additional attributes or markup depending on the (future) providers, e.g. stuff like `allow="autoplay"` or `allowfullscreen`. See my comment on `ALLOWED_PROVIDERS` above.

Reply:
This could be added later if need be